### PR TITLE
Refactor/pid registry

### PR DESCRIPTION
**Pull request by ankur22**
### Description of changes

This moves the `pidRegistry` from the `browser` package into the `registry` package. This refactor will help isolate the details to within the `registry` package.

Related to: https://github.com/grafana/xk6-browser/pull/889#discussion_r1200622188

### Checklist
```[tasklist]
- [X] Written tests for the changes
```

### CODE REVIEW

File: tests/webvital_test.go [hunk 1]
```diff
@@ -20,16 +20,16 @@ func TestWebVitalMetric(t *testing.T) {
...
-			"webvital_time_to_first_byte_good":      false,
-			"webvital_first_contentful_paint":       false,
-			"webvital_first_contentful_paint_good":  false,
-			"webvital_largest_content_paint":        false,
-			"webvital_largest_content_paint_good":   false,
-			"webvital_first_input_delay":            false,
-			"webvital_first_input_delay_good":       false,
-			"webvital_cumulative_layout_shift":      false,
-			"webvital_cumulative_layout_shift_good": false,
+			"browser_web_vital_ttfb":      false,
```
Comment (inancgumus) [inline tests/webvital_test.go]: Are these changes related to this PR?

**Review by inancgumus — APPROVED** [review]
LGTM besides some naming suggestions 👍 Personally, I would rename the `browser` package to `module` and put all these in a type called `Registry` in the `browser` package. —as they are closely related, these are module _globals_. Is there a reason we're separating them into different packages?

**Comment by ankur22** [comment]
> In my opinion, it would be better to rename the browser package to module and consolidate these "registries" under a type called Registry in its own file registry.go within the module package. Reads fine: module.Registry. Since they're closely related and contain module globals, it makes sense to streamline them into one package. Is there a specific reason for keeping them separate?

This is a good suggestion. I think we may have more "registries" that we need to create such as `subprocessRegistry`, `connectionRegistry` (or Pool), `browserRegistry` etc. I think they'll need to span the lifecycle of the test run 🤔 Maybe worth refactoring this again in another PR once we start working on those other registries? 🙂

File: registry/pid.go [hunk 2]
```diff
@@ -1,23 +1,23 @@
-package browser
+package registry
 
 import "sync"
 
-// pidRegistry keeps track of the launched browser process IDs.
-type pidRegistry struct {
+// PidRegistry keeps track of the launched browser process IDs.
+type PidRegistry struct {
```
[the remark this answers is not in the document]
Comment (ankur22) [reply registry/pid.go:6]: Yeah, I was thinking about that too. Would `registry.Pid` be better? I think for now it might be ok, considering we have things like `context.BackgroundContext`. As you've pointed out another good suggestion, we might need to refactor this again very soon, so i might leave the name as it until that next refactor 🙂

File: registry/remote.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	"crypto/rand"
+	"math/big"
+	"strings"
+
+	"github.com/grafana/xk6-browser/env"
+)
+
+// RemoteRegistry contains the details of the remote web browsers.
+// At the moment it's the WS URLs.
+type RemoteRegistry struct {
```
[the remark this answers is not in the document]
Comment (ankur22) [reply registry/remote.go]: Look here: https://github.com/grafana/xk6-browser/pull/899#discussion_r1201695622

**Comment by ka3de** [comment]
> In my opinion, it would be better to rename the browser package to module and consolidate these "registries" under a type called Registry in its own file registry.go within the module package. Reads fine: module.Registry. Since they're closely related and contain module globals, it makes sense to streamline them into one package. Is there a specific reason for keeping them separate?

I like this idea also :+1:

**Comment by ankur22** [comment]
I'm going to merge this in and work on the refactor that Inanc suggested. It shouldn't take long 🤞
